github/resource_github_branch_protection_v3_utils.go

```diff
@@ -46,10 +46,24 @@ func flattenAndSetRequiredStatusChecks(d *schema.ResourceData, protection *githu
 			contexts = append(contexts, c)
 		}
 
+		var checks []interface{}
+		for _, chk := range rsc.Checks {
+			chkMap := make(map[string]interface{})
+			chkMap["context"] = chk.Context
+			chkMap["app_id"] = func() int {
+				if chk.AppID != nil {
```

Comment:
If this is the ID from upstream via the SDK, it's likely to already be defaulted to a non-nil making this check redundant.